package.json

```diff
@@ -17,7 +17,9 @@
     "dist/testing",
     "dist/util",
     "dist/index.js",
-    "dist/index.d.ts"
+    "dist/index.d.ts",
+    "testing.d.ts",
+    "testing.js"
```

Comment:
Is it possible to put `testing` inside of the `dist` directory?

Reply:
TLDR; No, under `dist` doesn't work, but I think the latest commit is cleaner. Have a look.

More info...

Putting it under `dist` was actually my first attempt. Didn't work. So I went and looked at the [Typescript docs for module resolution](https://www.typescriptlang.org/docs/handbook/module-resolution.html).

Here is a synopsis:

The rules for module resolution of `import { x } from 'Foo'` are basically:

1. node_modules/Foo.(ts | tsx | d.ts)
2. node_modules/Foo/package.json w/ "types" specified
3. node_modules/Foo/index.(ts | tsx | d.ts)

Similar in node-land, but looking for 'js' files and package.json with "main" specified.

The `import { component } from '@stencil/core'` uses item number 2 for resolution (`node_modules/@stencil/core` has a `package.json` with "main" and "types" specified, both pointing to files in `dist/`).

So for resolving `@stencil/core/testing` I can do one of the following:

1. `node_modules/@stencil/core/testing.d.ts`
2. `node_modules/@stencil/core/testing/package.json` with "main" and "types" specified
3. `node_modules/@stencil/core/testing/index.d.ts` (plus .js for node land)

See my latest commit where I switched from strategy 1 to strategy 3 above. I think that is cleaner.

Reply:
Thinking about this more, I think option number 2 may make this more maintainable. Let me have a stab at that and see how it looks before we do anything else with this PR.